### PR TITLE
lib: trivial spelling fixes

### PR DESCRIPTION
**Pull request by tsaeger**
###### Motivation for this change

Noticed spelling errors in lib file, then found as many other spelling errors in lib.

###### Things done

1. Found sorted list of "bad" words from aspell.

    cat lib/*.nix |aspell list |sort |uniq -c |sort -nr > spells.txt

2. Went through the generated file and found a few more spelling errors in lib.

- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] Linux
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: lib/lists.nix [hunk 1]
```diff
@@ -148,7 +148,7 @@ rec {
     let found = filter pred list;
     in if found == [] then default else head found;
 
-  /* Return true iff function `pred' returns true for at least element
```
Comment (joachifm) [inline lib/lists.nix]: "iff" is a short-hand for "if and only if" (using just "if" here suggests that the function could return true in other cases as well)

**Comment by joachifm** [comment]
Having scanned it briefly, looks okay to me apart from the `iff -> if` change, which is incorrect.

**Comment by tsaeger** [comment]
Ok I will fix the iff's



On Apr 19, 2017 5:28 PM, "Joachim F." <notifications@github.com> wrote:

Having scanned it briefly, looks okay to me apart from the iff -> if
change, which is incorrect.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/NixOS/nixpkgs/pull/25031#issuecomment-295473360>, or mute
the thread

**Comment by tsaeger** [comment]
`iff` has been restored in the latest.

**Comment by Mic92** [comment]
Thanks!

**Pull request merged**